### PR TITLE
[entt] add `experimental` feature

### PR DESCRIPTION
**Pull request by TerensTare**
**Describe the pull request**

- #### What does your PR fix?  
  This PR adds a new `experimental` feature for [EnTT](https://github.com/skypjack/entt), which gets the source code from the `experimental` branch of the repo instead of `master`.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All triplets should work correctly. I have not updated the CI baseline.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  My PR follows the maintainer guide.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  I ran the command and it didn't show any output.

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@TerensTare ,Thanks for your pr.
Please run command "./vcpkg x-add-version entt --overwrite-version" then commit the change again

**Comment by TerensTare** [comment]
@JonLiu1993 thanks for responding. This is my first time working on a port and I'm not sure if I have correctly used `x-add-version`, so please let me know if this is the case.

**Comment by JonLiu1993** [comment]
> @JonLiu1993 thanks for responding. This is my first time working on a port and I'm not sure if I have correctly used `x-add-version`, so please let me know if this is the case.

You used it correctly, thank you

**Comment by JonLiu1993** [comment]
Feature is tested successfully in the following triplet:
- x64-windows
- x64-windows-static
- x64-linux

File: ports/entt/portfile.cmake [hunk 1]
```diff
@@ -1,10 +1,19 @@
-vcpkg_from_github(
-    OUT_SOURCE_PATH SOURCE_PATH
-    REPO skypjack/entt
-    REF 5d15a3d69f433a0e3fce266caaeb87c77c10453c #v3.7.1
-    SHA512 dbfb141dd706e1552a81e01005a28e916b369f29c0adfff337799f8375a9676e60f620b7981633829d6d175297088ace58e3c16cc802ab9d71681efebb1caba6
-    HEAD_REF master
-)
+if ("experimental" IN_LIST FEATURES)
+    set(VCPKG_USE_HEAD_VERSION ON)
```
Comment (ras0219-msft) [inline ports/entt/portfile.cmake:2]: Ports in the curated catalog are not allowed to set `VCPKG_USE_HEAD_VERSION` in the portfile. By design, there is no way for features and ports to have dynamic behavior without the user explicitly setting `--head` on the command line.

There are a few alternative options we can consider:

1. Just remove this line and require the user to also pass `--head`.
2. Supply a proper REF+SHA512 from the experimental branch.
3. Use a local ports overlay to get this behavior. In this case, no PR is needed; you can have this behavior locally for your project by checking in this replacement portfile and passing it to `--overlay-ports=` or `VCPKG_OVERLAY_PORTS`

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: ports/entt/portfile.cmake (hunk 1 above)
Comment (TerensTare) [reply ports/entt/portfile.cmake:2]: Point 1 is more adapt to the port, given the `experimental` branch tends to change continuously. I am commiting the changes to support alternative 1.

**Review by PhoebeHui — APPROVED** [review]

**Review by vicroms — APPROVED** [review]
Need to fix merge conflicts, LGTM otherwise.

**Pull request merged**